File: JAVA/200-299/231_PowerOfTwo.java

// Given an integer, write a function to determine if it is a power of two.

//Idea: An integer n is power of two iff n & (n-1) == 0. 
//For example: 16 -> 10000, 16 - 1 = 01111, 10000 & 01111 = 00000.  

public class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n < 1) return false;
        return (0 == (n & (n-1)));
    }
}

File: PYTHON/200-299/231_Power_of_Two.py
#
#Given an integer, write a function to determine if it is a power of two.
#
#
#Credits:Special thanks to @jianchao.li.fighter for adding this problem and creating all test cases.

class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        if n == 0: return False
        return not n & (n - 1)